protocol/messenger_communities.go

```diff
@@ -3168,6 +3168,17 @@ importMessageArchivesLoop:
 				return err
 			}
 
+			community, err := m.communitiesManager.GetByID(communityID)
+			if err != nil {
+				m.communitiesManager.LogStdout("couldn't get community to import archives", zap.Error(err))
+				return err
+			}
+
+			if !community.HasMember(&m.identity.PublicKey) {
+				m.communitiesManager.LogStdout("can't import archives when user not a member of community")
+				break importMessageArchivesLoop
+			}
```

Comment:
I think generally the check is fine, but I think it should not happen inside of this loop. We know the communityID before we run into this part here, so I think we shouldn't even call `importHistoryArchives()` if we know we aren't a member of the community.

If, however, we are a member, the loop is running and our memership state changes to not being a member, then we need to ensure that all tasks related to this community are stopped.

Reply:
Good point! thanks!